### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by davidcavazos**
Initial commit on Go starter project.

R: @aaltay @pcoet @lostluck 

GitHub Actions tests are passing in my fork: https://github.com/davidcavazos/beam-starter-go/actions

### CODE REVIEW

File: main.go
Comment (lostluck) [inline main.go]: Type definitions don't use an `=`. Using an equal defines a type alias, which isn't what we want here.

```suggestion
type Result struct {
```

File: main.go
Comment (lostluck) [inline main.go]: This should be called in `main`, right after flag parsing.

File: main.go
Comment (lostluck) [inline main.go]: While an anonymous function *can* work in Beam Go, it will cause more trouble than it's worth. Move it to a standard definition, and use an init block to register the function.

Otherwise, folks moving beyond the direct runner will have a bad time.

```
func printAndEmit(elem string, emit(string)) { 
  fmt.Println(elem)
  emit(elem) 
}

func init() {
  // DoFns should be registered with Beam.
  beam.RegisterFunction(printAndEmit)
}
```

File: main.go
Comment (lostluck) [reply main.go]: Secondly, while one *can* define a type for this, one shouldn't. Go would just use multiple returns. 

In this case, I would recommend calling `pipeline, scope := beam.NewPipelineWithRoot()` in `main`, and passing in the scope variable. Then we simply return the PCollection.

**Review by lostluck — CHANGES_REQUESTED** [review]

File: main.go
Comment (davidcavazos) [reply main.go]: Got it, that's a great suggestion!

File: main.go
Comment (davidcavazos) [reply main.go]: Sounds good, moved.

File: main.go
Comment (davidcavazos) [reply main.go]: That's good to know! Done.

File: CONTRIBUTING.md
Comment (pcoet) [inline CONTRIBUTING.md]: "contribute, here" -> "contribute. Here"

File: CONTRIBUTING.md
Comment (pcoet) [inline CONTRIBUTING.md]: "see to the" -> "see the"

**Review by pcoet — APPROVED** [review]

File: main.go
Comment (lostluck) [inline main.go]: import the Beam [log package](https://pkg.go.dev/github.com/apache/beam/sdks/v2/go/pkg/beam/log), and use log.Infoln instead of fmt.Println.  This allows the print to be visible from distributed / portable runners. fmt.Println will simply have the output lost to the worker container.

File: main.go
Comment (lostluck) [inline main.go]: Style wise, this is a code smell. Simpler to simply return without the variable.
```suggestion
	return beam.ParDo(scope, printAndEmit, elements)
```

File: .github/workflows/test.yaml
Comment (lostluck) [inline .github/workflows/test.yaml]: Reduce maintenance for us and users, and have the version pulled from the go.mod file: https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file
So for this it's probably `go-version-file: 'go.mod'`

File: main.go
Comment (lostluck) [inline main.go]: Since we aren't using the return, we can simply avoid mentioning it here.
```suggestion
	my_pipeline(scope, *input_text)
```

File: main.go
Comment (lostluck) [inline main.go]:
```suggestion
func myPipeline(scope beam.Scope, input_text string) beam.PCollection {
```
Go style has methods/functions use camelcase rather than snake case (snake case is for flags).

**Review by lostluck — CHANGES_REQUESTED** [review]
I think this will be the last of the nits from me. Very exciting!

File: main.go
Comment (davidcavazos) [reply main.go]: Sounds good, I created an extra function to show how to do a "simple" function and a function that takes a Context and emits values.

File: .github/workflows/test.yaml
Comment (davidcavazos) [reply .github/workflows/test.yaml]: Good catch, thanks!

**Review by lostluck — APPROVED** [review]
LGTM

**Comment by lostluck** [comment]
Since this is in the apache org, I can merge this if you can't. Please let me know, and it'll be done.

**Comment by davidcavazos** [comment]
Hi @lostluck, I can't merge it. It would be great if you helped us merge it :)

**Comment by lostluck** [comment]
Merged!

**Comment by brucearctor** [comment]
Wohooo!  Congrats!

**Pull request merged**
